junit-jupiter-api/src/main/java/org/junit/jupiter/api/DisplayNameGenerator.java

```diff
@@ -141,4 +141,100 @@ private static boolean hasParameters(Method method) {
 		}
 	}
 
+	/**
+	 * {@code DisplayNameGenerator} that supports {@code ReplaceUnderscores} to generate complete sentences.
+	 *
+	 * <p>This generator extends the functionality of {@link ReplaceUnderscores} by
+	 * generating a human-readable display names that form complete sentences divided each
+	 * class, nested class and test by a ({@code ','}).
+	 *
+	 * @since 5.6
+	 */
+	@API(status = EXPERIMENTAL, since = "5.6")
+	class IndicativeSentencesGenerator extends ReplaceUnderscores {
+
+		@Override
+		public String generateDisplayNameForClass(Class<?> testClass) {
+			return removeRootCharacter(super.generateDisplayNameForClass(testClass));
+		}
+
+		@Override
+		public String generateDisplayNameForNestedClass(Class<?> nestedClass) {
+			return classReplaceToIndicativeSentence(nestedClass);
+		}
+
+		@Override
+		public String generateDisplayNameForMethod(Class<?> testClass, Method testMethod) {
+			return classReplaceToIndicativeSentence(testClass) + getSentenceSeparator(testClass)
+					+ super.generateDisplayNameForMethod(testClass, testMethod);
+		}
+
+		private String classReplaceToIndicativeSentence(Class<?> testClass) {
+			Class<?> classWithEnclosingParent = testClass.getEnclosingClass();
+			DisplayName classWithDisplayName = testClass.getAnnotation(DisplayName.class);
+			DisplayNameGeneration classWithAnnotation = testClass.getAnnotation(DisplayNameGeneration.class);
+
+			if (classWithEnclosingParent == null) {
+				if (classWithDisplayName != null)
+					return classWithDisplayName.value();
+				else
+					return generateDisplayNameForClass(testClass);
+			}
+			else {
+				if (classWithAnnotation != null && classWithAnnotation.value() == IndicativeSentencesGenerator.class) {
+					if (classWithDisplayName != null)
+						return classWithDisplayName.value();
+					else
+						return generateDisplayNameForClass(testClass);
+				}
+				else {
+					if (classWithDisplayName != null)
+						return classReplaceToIndicativeSentence(classWithEnclosingParent)
+								+ getSentenceSeparator(testClass) + classWithDisplayName.value();
+					else
+						return classReplaceToIndicativeSentence(classWithEnclosingParent)
+								+ getSentenceSeparator(testClass) + super.generateDisplayNameForNestedClass(testClass);
+
+				}
+			}
+		}
+
+		/**
+		 * Gets the separator for {@link IndicativeSentencesSeparator} when extracting the
+		 * annotation from {@code IndicativeSentencesSeparator}, if it doesn't find it,
+		 * then search for the parent classes, if no separator is found use @code{", "} by default.
+		 *
+		 * @param currentClass the Test Class the separator either custom or default
+		 * @return the indicative sentence separator
+		 * {@code Class.getName()}.
+		 */
+		private String getSentenceSeparator(Class<?> currentClass) {
+			IndicativeSentencesSeparator separator = currentClass.getAnnotation(IndicativeSentencesSeparator.class);
```

Comment:
You can also make use of `findAnnotation` here

Reply:
thanks @juliette-derancourt, done in the @marcphilipp refactoring suggestions :D